pkg/machines/vmnetworktab.jsx

```diff
@@ -0,0 +1,191 @@
+/*jshint esversion: 6 */
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import cockpit from 'cockpit';
+import { changeNetworkState } from "./actions.es6";
+import { rephraseUI, vmId } from "./helpers.es6";
+import { Listing, ListingRow } from 'cockpit-components-listing.jsx';
+
+const _ = cockpit.gettext;
+
+class VmNetworkTab extends React.Component {
+    constructor(props) {
+        super(props);
+        const that = this;
+        this.hostNetworks = [];
```

Comment:
I don't think you want to do this on each tab. I'd suggest watching on dbus at a higher level and having each tab access that list.

client.proxies might be a good way to do this.